src/liballoc/heap.rs

```diff
@@ -26,6 +26,10 @@ pub unsafe fn allocate(size: usize, align: usize) -> *mut u8 {
 ///
 /// On failure, return a null pointer and leave the original allocation intact.
 ///
+/// If the allocation was relocated, the memory at the passed-in pointer is undefined after the
+/// call.
+///
```

Comment:
stray double newline, I'd also tend to preserve the surrounding style by wrapping to 80 chars as well. Other than that though r=me